File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20047RecommendedWines.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import java.math.BigDecimal

/**
 * 
 * @param id 
 * @param title 
 * @param averageRating 
 * @param description 
 * @param imageUrl 
 * @param link 
 * @param price 
 * @param ratingCount 
 * @param score 
 */
case class InlineResponse20047RecommendedWines(id: Int,
                title: String,
                averageRating: BigDecimal,
                description: String,
                imageUrl: String,
                link: String,
                price: String,
                ratingCount: Int,
                score: BigDecimal
                )

object InlineResponse20047RecommendedWines {
    /**
     * Creates the codec for converting InlineResponse20047RecommendedWines from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20047RecommendedWines] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20047RecommendedWines] = deriveEncoder
}

File: clojure/src/spoonacular_api/specs/inline_response_200_47_recommended_wines.clj
(ns spoonacular-api.specs.inline-response-200-47-recommended-wines
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def inline-response-200-47-recommended-wines-data
  {
   (ds/req :id) int?
   (ds/req :title) string?
   (ds/req :averageRating) float?
   (ds/req :description) string?
   (ds/req :imageUrl) string?
   (ds/req :link) string?
   (ds/req :price) string?
   (ds/req :ratingCount) int?
   (ds/req :score) float?
   })

(def inline-response-200-47-recommended-wines-spec
  (ds/spec
    {:name ::inline-response-200-47-recommended-wines
     :spec inline-response-200-47-recommended-wines-data}))
